addons/mail/static/src/discuss/call/common/blur_performance_warning.js

```diff
@@ -0,0 +1,75 @@
+import { CallPopover } from "@mail/discuss/call/common/call_popover";
+
+import { Component, useState, useEffect } from "@odoo/owl";
+
+import { isMobileOS } from "@web/core/browser/feature_detection";
+import { useService } from "@web/core/utils/hooks";
+import { memoize } from "@web/core/utils/functions";
+import { _t } from "@web/core/l10n/translation";
+
+/**
+ * Checks if the browser supports hardware acceleration for video processing.
+ *
+ * @returns {boolean} True if hardware acceleration is supported, false otherwise.
+ */
+const checkHardwareAccelerationSupport = memoize(() => {
+    const canvas = document.createElement("canvas");
+    const gl =
+        canvas.getContext("webgl2") ||
+        canvas.getContext("webgl") ||
+        canvas.getContext("experimental-webgl");
+    if (!gl) {
+        // WebGL support is typically required for hardware acceleration.
+        return false;
+    }
+    const debugInfo = gl.getExtension("WEBGL_debug_renderer_info");
+    if (debugInfo) {
+        const renderer = gl.getParameter(debugInfo.UNMASKED_RENDERER_WEBGL);
+        if (/swiftshader|llvmpipe|software/i.test(renderer)) {
+            // These renderers indicate software-based rendering instead of hardware acceleration.
+            return false;
+        }
+    }
+    return true;
+});
+
+export class BlurPerformanceWarning extends Component {
+    static template = "discuss.BlurPerformanceWarning";
+    static props = {};
+    static components = { CallPopover };
+
+    setup() {
+        this.rtc = useService("discuss.rtc");
```

Comment:
Before this commit and with this commit, when I disable hardware acceleration on my browser (Edge), I got a browser warning from failing to paint on canvas + traceback, in addition to this heuristic code.

Not a big deal but I wonder if the heuristic covers more and/or there's necessarily the traceback with it.

<img width="462" height="148" alt="Screenshot 2025-07-10 at 15 27 49" src="https://github.com/user-attachments/assets/7d9d9056-b581-4eef-869f-86aa0a97cf5f" />